### PR TITLE
DOC: add layout='none' option to Figure constructor

### LINKED ISSUE #25223: [Doc]: `layout="none"` for figure constructor?
### Documentation Link

https://matplotlib.org/devdocs/api/_as_gen/matplotlib.pyplot.figure.html

### Problem

Suppose you have constrained layout turned on in general via `rcParams`, but want to not use it for a particular figure.  This seems to work

```python
import matplotlib.pyplot as plt

plt.rcParams["figure.constrained_layout.use"] = True
fig = plt.figure(layout="none")
print(fig.get_layout_engine())
```
```
None
```
but `layout="none"` is not documented as a valid option in either `pyplot.figure` or the [Figure](https://matplotlib.org/devdocs/api/figure_api.html#matplotlib.figure.Figure) constructor.

### Suggested improvement

Should  `layout="none"`  be documented as a valid option?

### PR DESCRIPTION
**Pull request by rcomer**
## PR Summary

Closes #25223.  Documents the `layout='none'` option for creating a figure.

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

**Review by jklymak — APPROVED** [review]
This is fine.  Note that for pyplot it's not great to refer to the .Figure constructor for the meaning of the layout options.

**Comment by rcomer** [comment]
I could copy the full description across to pyplot?

**Comment by jklymak** [comment]
I would, but it could be a different PR.

File: lib/matplotlib/pyplot.py [hunk 1]
```diff
@@ -740,13 +740,38 @@ def figure(num=None,  # autoincrement if None, else integer from 1-N
...
-    layout : {'constrained', 'tight', 'compressed', \
-        `.LayoutEngine`, None}, default: None
+    layout : {'constrained', 'compressed', 'tight', 'none', `.LayoutEngine`, None}, \
+default: None
         The layout mechanism for positioning of plot elements to avoid
         overlapping Axes decorations (labels, ticks, etc). Note that layout
-        managers can measurably slow down figure display. Defaults to *None*
-        (but see the documentation of the `.Figure` constructor regarding the
-        interaction with rcParams).
+        managers can measurably slow down figure display. Defaults to *None*.
```
Comment (QuLogic) [inline lib/matplotlib/pyplot.py:747]: Probably can remove the Default here as it's in the type line.

File: lib/matplotlib/pyplot.py [hunk 2]
```diff
@@ -740,13 +740,38 @@ def figure(num=None,  # autoincrement if None, else integer from 1-N
...
+            simple algorithm that adjusts the subplot parameters so that
+            decorations do not overlap. See `.Figure.set_tight_layout` for
+            further details.
+
+        - 'none': Do not use a layout engine.
+
+        - A `.LayoutEngine` instance. Builtin layout classes are
+            `.ConstrainedLayoutEngine` and `.TightLayoutEngine`, more easily
+            accessible by 'constrained' and 'tight'.  Passing an instance
+            allows third parties to provide their own layout engine.
```
Comment (QuLogic) [inline lib/matplotlib/pyplot.py:770]: There is an extra indent on all the continuation lines of each bullet point.

Comment (rcomer) [reply lib/matplotlib/pyplot.py:770]: oooops

File: lib/matplotlib/pyplot.py (hunk 1 above)
Comment (rcomer) [reply lib/matplotlib/pyplot.py]: Done.

File: lib/matplotlib/pyplot.py (hunk 2 above)
Comment (rcomer) [reply lib/matplotlib/pyplot.py]: Fixed.

**Comment by jklymak** [comment]
https://output.circle-artifacts.com/output/job/4e86f311-ca8d-4293-907a-1870d985995e/artifacts/0/doc/build/html/api/figure_api.html#matplotlib.figure.Figure

Looks like the formatting still needs some work?

File: lib/matplotlib/figure.py [hunk 3]
```diff
@@ -2420,11 +2420,11 @@ def __init__(self,
                 The use of this parameter is discouraged. Please use
                 ``layout='constrained'`` instead.
 
-        layout : {'constrained', 'compressed', 'tight', `.LayoutEngine`, None}
+        layout : {'constrained', 'compressed', 'tight', 'none', `.LayoutEngine`, \
+None}, default: None
```
Comment (jklymak) [inline lib/matplotlib/figure.py:2424]: Its not your fault, but the `'constrained'` bullet below has an extra indent

Comment (rcomer) [reply lib/matplotlib/figure.py:2424]: Oh _that's_ why it's in bold.  I was scratching my head about that...

**Comment by rcomer** [comment]
I think it's OK now
* [pyplot.figure](https://output.circle-artifacts.com/output/job/384b2120-1c37-41dc-b079-9a03c18a3df2/artifacts/0/doc/build/html/api/_as_gen/matplotlib.pyplot.figure.html#matplotlib-pyplot-figure)
* [Figure](https://output.circle-artifacts.com/output/job/384b2120-1c37-41dc-b079-9a03c18a3df2/artifacts/0/doc/build/html/api/figure_api.html#matplotlib.figure.Figure)

**Review by jklymak — APPROVED** [review]

**Comment by jklymak** [comment]
Agreed, thanks for cleaning up the previous breakage!

**Pull request merged**